utils/node_local_test.py

```diff
@@ -3231,6 +3233,144 @@ def test_daos_fs_check(self):
         if dir_nr != 12:
             raise NLTestFail('Wrong number of sub-directories in lost+found')
 
+    def test_daos_fs_fix(self):
+        """Test DAOS FS Fix Tool"""
+
+        pool = self.pool.id()
+        cont = self.container
+
+        dfuse = DFuse(self.server,
+                      self.conf,
+                      pool=self.pool.id(),
+                      container=self.container,
+                      caching=False)
+        dfuse.start(v_hint='fs_fix_test', log_mask='CRIT')
+        path = dfuse.dir
+        dirname = join(path, 'test_dir')
+        os.mkdir(dirname)
+
+        fname1 = join(dirname, 'f1')
+        with open(fname1, 'w') as fd:
+            fd.write('test1')
+        fname2 = join(dirname, 'f2')
+        with open(fname2, 'w') as fd:
+            fd.write('test2')
+
+        dirname1 = join(path, 'test_dir/1d1/')
+        os.mkdir(dirname1)
+        fname3 = join(dirname1, 'f3')
+        with open(fname3, 'w') as fd:
+            fd.write('test3')
+        dirname2 = join(path, 'test_dir/1d2/')
+        os.mkdir(dirname2)
+        fname4 = join(dirname2, 'f4')
+        with open(fname4, 'w') as fd:
+            fd.write('test4')
+
+        # start corrupting things
+        daos_mw_fi = join(self.conf['PREFIX'], 'lib/daos/TESTING/tests/', 'daos_mw_fi')
+        cmd_env = get_base_env()
+        cmd_env['DAOS_AGENT_DRPC_DIR'] = self.conf.agent_dir
+        cmd = [daos_mw_fi, pool, cont, "corrupt_entry", "/test_dir/f1"]
+        print(cmd)
+        rc = subprocess.run(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE,
+                            env=cmd_env, check=False, cwd=dfuse.dir)
+        assert rc.returncode == 0
+        cmd = [daos_mw_fi, pool, cont, "corrupt_entry", "/test_dir/1d1/f3"]
+        print(cmd)
+        rc = subprocess.run(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE,
+                            env=cmd_env, check=False, cwd=dfuse.dir)
+        assert rc.returncode == 0
+        cmd = [daos_mw_fi, pool, cont, "corrupt_entry", "/test_dir/1d2"]
+        print(cmd)
+        rc = subprocess.run(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE,
+                            env=cmd_env, check=False, cwd=dfuse.dir)
+        assert rc.returncode == 0
+
+        # try to read from corrupted entries. all should fail
+        try:
+            fd = os.open(fname1, 'r')
+            os.close(fd)
+            assert False
+        except TypeError:
+            pass
+
+        try:
+            fd = os.open(fname3, 'r')
+            os.close(fd)
+            assert False
+        except TypeError:
+            pass
+
+        try:
+            dir_list = os.listdir(dirname2)
+            assert False
+        except OSError:
```

Comment:
FYI - OSError is a generic "the kernel gave an errno" error, you may be able to trap on a subtype of this to make it more resilient.
```suggestion
        except OSError as error:
            assert error.errno == errno.???
```
Search for errno. elsewhere in this file to see what I mean.